src/model.jl

```diff
@@ -499,89 +549,120 @@ function set_capacity_cost(m, n, 𝒯, t_inv,  modeltype::EnergyModel, ::Rolling
 end
 
 #same function dispatched for storages
-function set_capacity_cost(m, n::Storage, 𝒯, t_inv,  modeltype::EnergyModel, ::UnlimitedLife)
+
+set_capacity_cost(m, n, inv_data, prefix, 𝒯ᴵⁿᵛ, modeltype) =
+    set_capacity_cost(m, n, inv_data, prefix, 𝒯ᴵⁿᵛ, modeltype, lifetime_mode(inv_data))
+function set_capacity_cost(m, n, inv_data, prefix, 𝒯ᴵⁿᵛ, modeltype, ::UnlimitedLife)
+    # Deduce the required variables
+    var_capex = get_var_capex(m, prefix, n)
+    var_rem = get_var_rem(m, prefix, n)
+
     # The capacity has an unlimited lifetime, one investment at the beginning of t_inv
-    capex_val = capex(n, t_inv)
-    @constraint(m, m[:capex_stor][n, t_inv] == capex_val.level * m[:stor_cap_add][n, t_inv])
-    @constraint(m, m[:capex_rate][n, t_inv] == capex_val.rate * m[:stor_rate_add][n, t_inv])
-    @constraint(m, m[:stor_cap_rem][n, t_inv] == 0)
-    @constraint(m, m[:stor_rate_rem][n, t_inv] == 0)
+    capex_val = set_capex_value(m, n, inv_data, prefix, 𝒯ᴵⁿᵛ)
+    @constraint(m, [t_inv ∈ 𝒯ᴵⁿᵛ], var_capex[t_inv] == capex_val[t_inv])
+
+    # Fix the binary variable
+    for t_inv ∈ 𝒯ᴵⁿᵛ
+        fix(var_rem[t_inv], 0; force=true)
+    end
 end
 
-function set_capacity_cost(m, n::Storage, 𝒯, t_inv,  modeltype::EnergyModel, ::StudyLife)
+function set_capacity_cost(m, n, inv_data, prefix, 𝒯ᴵⁿᵛ, modeltype, ::StudyLife)
+    # Deduce the required variables
+    var_capex = get_var_capex(m, prefix, n)
+    var_rem = get_var_rem(m, prefix, n)
+
     # The capacity is limited to the end of the study. Reinvestments are included
     # No capacity removed as there are reinvestments according to the study length
-    capex_val = capex(n, t_inv)
-    stor_capex = capex_val.level * set_capex_value(remaining(t_inv, 𝒯), lifetime(n, t_inv), discount_rate(modeltype))
-    rate_capex = capex_val.rate * set_capex_value(remaining(t_inv, 𝒯), lifetime(n, t_inv), discount_rate(modeltype))
-    @constraint(m, m[:capex_stor][n, t_inv] == stor_capex * m[:stor_cap_add][n, t_inv])
-    @constraint(m, m[:capex_rate][n, t_inv] == rate_capex * m[:stor_rate_add][n, t_inv])
-    @constraint(m, m[:stor_cap_rem][n, t_inv] == 0)
-    @constraint(m, m[:stor_rate_rem][n, t_inv] == 0)
+    capex_disc = StrategicProfile(
+        [
+            set_capex_discounter(
+                remaining(t_inv, 𝒯ᴵⁿᵛ),
+                lifetime(inv_data, t_inv),
+                discount_rate(modeltype)
+            ) for t_inv ∈ 𝒯ᴵⁿᵛ
+        ]
+    )
+    capex_val = set_capex_value(m, n, inv_data, prefix, 𝒯ᴵⁿᵛ)
+    @constraint(m, [t_inv ∈ 𝒯ᴵⁿᵛ], var_capex[t_inv] == capex_val[t_inv] * capex_disc[t_inv])
+
+    # Fix the binary variable
+    for t_inv ∈ 𝒯ᴵⁿᵛ
```

Comment:
I have to check it, I will come back to you.

Reply:
You mean removing the fixing of the removal variable? Sure, we could. In practice, I think that it is actually a smart move, as it would allow the model to retire capacity early in cases it is not necessary.

Note, that I just did a one to one translation of what was there beforehand.

Reply:
What I meant was if we can avoid constructing the variables and constraints for removal instead of fixing the variables.

I guess we would want to have different options for whether technologies may or may not be retired early (or in some cases must retire under certain conditions) We can come back to this later.